cparser/Elab.ml

```diff
@@ -1593,24 +1637,25 @@ let elab_expr vararg loc env a =
       | TPtr(ty, _) | TArray(ty, _, _) ->
           { edesc = EUnop(Oderef, b1); etyp = ty },env
       | _ ->
-          error "argument of unary '*' is not a pointer"
+          error "indirection requires pointer operand (%a invalid)"
```

Comment:
The original message is clearer.  "indirection requires pointer operand" is obscure and pedantic.